openmc/deplete/microxs.py

```diff
@@ -192,8 +210,118 @@ def __init__(self, data: np.ndarray, nuclides: List[str], reactions: List[str]):
+            Path to the depletion chain XML file that will be used in
+            depletion simulation.
+        openmc_data_path : str, optional
+            Path to the cross section XML file that contains data library paths.
+        temperature : int, optional
+            Temperature for cross section evaluation in [K].
+
+        Returns
+        -------
+        MicroXS
+        """
+
+        check_type("temperature", temperature, int)
+        # if energy is string then use group structure of that name
+        if isinstance(energies, str):
+            energies = openmc.EnergyFilter.from_group_structure(energies).values
+        else:
+            # if user inputs energies check they are ascending (low to high) as some
+            # depletion codes use high energy to low energy.
+            if not all(energies[i] <= energies[i+1] for i in range(len(energies) - 1)):
+                raise ValueError('Energy bin must be in ascending order')
+
+        # check dimension consistency
+        if not len(multi_group_flux) == len(energies)-1:
+            raise ValueError('Length of flux array should be len(energies)-1')
+
+        chain_file_path = _resolve_chain_file_path(chain_file)
+        chain = openmc.deplete.Chain.from_xml(chain_file_path)
+
+        openmc_data_path = _resolve_openmc_data_path(openmc_data_path)
+        data_lib = openmc.data.DataLibrary.from_xml(openmc_data_path)
+
+        # get reactions and nuclides from chain file
+        nuclides, reactions = chain.nuclides, chain.reactions
+        nuclides = [nuc.name for nuc in nuclides]
+        if 'fission' in reactions:
+            # send to back
+            reactions.append(reactions.pop(reactions.index('fission')))
+        # convert reactions to mt file
+        #! I feel like this zero indexing will backfire
+        #! There has to be a better way for this
+        mts = [list(REACTIONS[reaction].mts)[0] for reaction in reactions if reaction != 'fission']
+        if 'fission' in reactions:
+            mts.append(18) # fission is not in the REACTIONS
+
+        microxs_arr = np.zeros((len(nuclides), len(mts)))
+
+        with TemporaryDirectory():
+            from ctypes import c_int, c_double
+
+            mat_all_nucs = openmc.Material()
+            for nuc in nuclides:
+                mat_lib = data_lib.get_by_material(nuc, data_type="neutron")
+                # limits the material to nuclides that have cross sections
+                if mat_lib:
+                    mat_all_nucs.add_nuclide(nuc, 1)
+            mat_all_nucs.set_density("atom/b-cm", 1)
+            materials = openmc.Materials([mat_all_nucs])
+
+            surf1 = openmc.Sphere(r=1, boundary_type="vacuum")
+            surf1_region = -surf1
+            surf1_cell = openmc.Cell(region=surf1_region)
+            geometry = openmc.Geometry([surf1_cell])
+
+            settings = openmc.Settings()
+            settings.particles = 1
+            settings.batches = 1
+
```

Comment:
I noticed this currently writes a summary.h5 file that takes a tiny amount of time and we don't need the file. So with this change we can avoid writing it :-)
```suggestion
            settings.output = {'summary': False}
```